### PR TITLE
feat(DMS): rocketmq supports specification modification

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:
  rocketmq supports specification modification
**Special notes for your reviewer**:

**Release note**:

```release-note
rocketmq supports specification modification: flavor_id, storage_space, broker_num and publicip(for increased brokers)
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./loud/services/acceptance/dms/ TESTARGS='-run TestAccDmsRocketMQInstance'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsRocketMQInstance -timeout 360m -parallel 4
=== RUN   TestAccDmsRocketMQInstance_basic
=== PAUSE TestAccDmsRocketMQInstance_basic
=== RUN   TestAccDmsRocketMQInstance_prepaid_basic
=== PAUSE TestAccDmsRocketMQInstance_prepaid_basic
=== RUN   TestAccDmsRocketMQInstance_broker_publicip
=== PAUSE TestAccDmsRocketMQInstance_broker_publicip
=== RUN   TestAccDmsRocketMQInstance_publicip
=== PAUSE TestAccDmsRocketMQInstance_publicip
=== CONT  TestAccDmsRocketMQInstance_basic
=== CONT  TestAccDmsRocketMQInstance_broker_publicip
=== CONT  TestAccDmsRocketMQInstance_publicip
=== CONT  TestAccDmsRocketMQInstance_prepaid_basic
--- PASS: TestAccDmsRocketMQInstance_publicip (704.74s)
--- PASS: TestAccDmsRocketMQInstance_prepaid_basic (1152.70s)
--- PASS: TestAccDmsRocketMQInstance_broker_publicip (1683.47s)
--- PASS: TestAccDmsRocketMQInstance_basic (3228.76s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       3228.806s
```

### CODE REVIEW

File: huaweicloud/common/common.go [hunk 1]
```diff
@@ -319,3 +319,20 @@ func RetryContextWithWaitForState(param *RetryContextWithWaitForStateParam) (int
 
 	return stateConf.WaitForStateContext(param.Ctx)
 }
+
+// GetEipsbyAddresses returns the EIPs of addresses when success.
+func GetEipsbyAddresses(client *golangsdk.ServiceClient, addresses []string) ([]eips.PublicIp, error) {
```
Comment (Jason-Zhang9309) [inline huaweicloud/common/common.go]: please check func GetEipIDAddress in common.go
the enterprise_project_id need to be take care

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_instance_test.go [hunk 2]
```diff
@@ -357,7 +571,7 @@ locals {
 
 resource "huaweicloud_dms_rocketmq_instance" "test" {
   name              = "%[2]s"
-  engine_version    = element(local.query_results.versions, length(local.query_results.versions)-1)
+  engine_version    = local.query_results.versions[1]
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_instance_test.go]: if you use index 1, that means the list has at least 2 elements.
are you sure about that?

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go [hunk 3]
```diff
@@ -879,3 +891,137 @@ func waitUpdatePublicipSuccess(ctx context.Context, d *schema.ResourceData, clie
...
+
+			resizeBodyParams["publicip_id"] = publicipIds
+		}
+
+		if err := doRocketmqInstanceResize(ctx, client, d, resizeBodyParams); err != nil {
+			return err
+		}
+	}
+
+	if d.HasChange("storage_space") {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go:927]: put this part at line 907, updating the broker_num should be the last step

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go [hunk 4]
```diff
@@ -879,3 +891,137 @@ func waitUpdatePublicipSuccess(ctx context.Context, d *schema.ResourceData, clie
...
+	for _, addressPort := range allAddressPortList {
+		addressList = append(addressList, strings.Split(addressPort, ":")[0])
+	}
+	publicips, err := common.GetEipsbyAddresses(eipClient, addressList)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ipIdList := make([]string, 0)
+	ipAddressList := make([]string, 0)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go:1008]:
```suggestion
	ipIdList := make([]string, len(publicips))
	ipAddressList := make([]string, len(publicips))
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go [hunk 5]
```diff
@@ -879,3 +891,137 @@ func waitUpdatePublicipSuccess(ctx context.Context, d *schema.ResourceData, clie
...
+	for _, ip := range publicips {
+		ipIdList = append(ipIdList, ip.ID)
+		ipAddressList = append(ipAddressList, ip.PublicAddress)
+	}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go]:
```suggestion
	ipIdList := make([]string, len(publicips))
	ipAddressList := make([]string, len(publicips))
	for i, ip := range publicips {
		ipIdList[i] = ip.ID
		ipAddressList[i] = ip.PublicAddress
	}
```
or like this:
```suggestion
	ipIdList := make([]string, 0, len(publicips))
	ipAddressList := make([]string, 0, len(publicips))
	for _, ip := range publicips {
		ipIdList = append(ipIdList, ip.ID)
		ipAddressList = append(ipAddressList, ip.PublicAddress)
	}
```

File: huaweicloud/common/common.go (hunk 1 above)
Comment (346010762) [reply huaweicloud/common/common.go]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_instance_test.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_instance_test.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go (hunk 3 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go:927]: storage_space is total storage space, there would be a changed plan if broker_num after storage_space.

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go (hunk 5 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go]: done

**Pull request merged**